### PR TITLE
feat: allow `persistent`? `actor` to switch default and `transient` as synonym for `flexible`

### PR DESCRIPTION
**Pull request by crusso**
This is the simplest solution with least grammar irregularities and probably least required doc changes.

* adds keywords `persistent` and `transient`.
* `transient` is a synonym for `flexible`. 
* `persistent? actor (class) ...` flips the default from `flexible` to `stable`.

Parses nicely because there is no any longer ambiguity between `stable`/`flexible` uses as object modifier or visibility. That means we don't need fancy and confusing grammar factoring.

### CODE REVIEW

File: test/run-drun/stable-counter/stable-counter-1.mo [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  transient let f = func(){};
+
+  func g() {};
+  class D() {};
+  type T = ?T;
+  ignore 1;
+
+}
+
+
```
Comment (crusso) [inline test/run-drun/stable-counter/stable-counter-1.mo:26]:
```suggestion
```

File: test/run-drun/stable-counter-class/stable-counter-class.mo [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  flexible let f = func(){};
+  func g() {};
+  class D() {};
+  type T = ?T;
+  ignore 1;
+
+}
+
+
```
Comment (crusso) [inline test/run-drun/stable-counter-class/stable-counter-class.mo:25]:
```suggestion
```

File: test/run-drun/stable-counter-class/stable-counter-class-2.mo [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  transient let f = func(){};
+  func g() {};
+  class D() {};
+  type T = ?T;
+  ignore 1;
+
+}
+
+
```
Comment (crusso) [inline test/run-drun/stable-counter-class/stable-counter-class-2.mo:25]:
```suggestion
```

File: test/run-drun/stable-counter-class/stable-counter-class-1.mo [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  transient let f = func(){};
+  func g() {};
+  class D() {};
+  type T = ?T;
+  ignore 1;
+
+}
+
+
```
Comment (crusso) [inline test/run-drun/stable-counter-class/stable-counter-class-1.mo:25]:
```suggestion
```

File: test/run-drun/stable-counter-class/stable-counter-class-2.mo [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  transient let f = func(){};
+  func g() {};
+  class D() {};
+  type T = ?T;
+  ignore 1;
+
+}
+
```
Comment (crusso) [inline test/run-drun/stable-counter-class/stable-counter-class-2.mo:24]:
```suggestion
```

File: src/mo_frontend/printers.ml [hunk 6]
```diff
@@ -193,7 +195,7 @@ let string_of_symbol = function
   | X (N N_path) -> "<path>"
   | X (N N_annot_opt) -> "<annot_opt>"
   | X (N N_seplist_case_semicolon_) ->  "seplist(<case>,<semicolon>)"
-  | X (N N_seplist_dec_SEMICOLON_) -> "seplist(<dec>,;)"
+  | X (N N_seplist_dec_SEMICOLON_) -> "seplist(<top_dec>,<semicolon>)"
```
Comment (crusso) [inline src/mo_frontend/printers.ml:198]:
```suggestion
  | X (N N_seplist_dec_SEMICOLON_) -> "seplist(<dec>,;)"
```

File: test/run-drun/stable-counter-class/stable-counter-class-2.mo [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import Prim "mo:⛔";
+
+persistent actor class Counter() {
+
+  var count : Nat = 0;
```
Comment (luc-blaeser) [inline test/run-drun/stable-counter-class/stable-counter-class-2.mo:5]: Maybe we should one more test with `transient`, even if uninteresting for testing, just for syntax coverage.

**Review by luc-blaeser — DISMISSED** [review]
Nice and simple PR. 
Let's get this merged!
Just two things that we can address in follow-up PRs:
1. We should also streamline documentation to this syntax. I'll ask @jessiemongeon1 
2. We could deprecate the old actor syntax with warning in a next step. Or ask users to explicitly specify stability modifier on old actor fields.

File: doc/md/examples/grammar.txt [hunk 8]
```diff
@@ -7,11 +7,16 @@
...
 
-<obj_sort> ::= 
+<typ_obj_sort> ::= 
     'object'
     'actor'
     'module'
 
+<obj_sort> ::= 
+    'object'
+    'persistent'? 'actor'
```
Comment (luc-blaeser) [inline doc/md/examples/grammar.txt:17]: Maybe a very brief mentioning in the docu would be good. We can then follow up with more cross-cutting docu updates later.

**Comment by luc-blaeser** [comment]
Thank you for this PR!

File: src/mo_frontend/parser.mly [hunk 9]
```diff
@@ -277,7 +290,7 @@ and objblock s id ty dec_fields =
 %type<Mo_def.Syntax.typ list> seplist(typ,COMMA)
 %type<Mo_def.Syntax.pat_field list> seplist(pat_field,semicolon)
 %type<Mo_def.Syntax.pat list> seplist(pat_bin,COMMA)
-%type<Mo_def.Syntax.dec list> seplist(imp,semicolon) seplist(imp,SEMICOLON) seplist(dec,semicolon) seplist(dec,SEMICOLON)
+%type<Mo_def.Syntax.dec list> seplist(imp,semicolon) seplist(imp,SEMICOLON) seplist(dec,semicolon)
```
[the remark this answers is not in the document]
Comment (crusso) [reply src/mo_frontend/parser.mly:293]: Oh no, I think you caught a bug.  I shouldn't be changing that bit of the grammar (might not matter though)

File: doc/md/examples/grammar.txt (hunk 8 above)
Comment (crusso) [reply doc/md/examples/grammar.txt:17]: Yeah, I'll do that.

**Comment by crusso** [comment]
> Nice and simple PR. Let's get this merged! Just two things that we can address in follow-up PRs:
> 
> 1. We should also streamline documentation to this syntax. I'll ask @jessiemongeon1
> 2. We could deprecate the old actor syntax with warning in a next step. Or ask users to explicitly specify stability modifier on old actor fields.

Maybe let me take a pass first before getting @jessiemongeon1 involved. She has tons on her plate and it might be quicker for me to revise the relevant bits.

File: src/mo_frontend/parser.mly (hunk 9 above)
Comment (crusso) [inline src/mo_frontend/parser.mly:293]:
```suggestion
%type<Mo_def.Syntax.dec list> seplist(imp,semicolon) seplist(imp,SEMICOLON) seplist(dec,semicolon) seplist(dec,SEMICOLON)
```

**Review by luc-blaeser — DISMISSED** [review]
Nice! Let's merge this.

**Review by luc-blaeser — APPROVED** [review]
Great doc overhaul!

File: Changelog.md [hunk 10]
```diff
@@ -1,5 +1,40 @@
...
 
+* motoko (`moc`)
+
+  * Breaking change (minor):
+
+    * Add new keyword `transient` with exactly the same meaning as the old keyword `flexible` (but a more familiar reading).
+
+    * Add keyword `persistent`.
+
+      When used to modify the `actor` keyword in an actor or actor class definition, the keyword declares that the default stability of an
```
Comment (ggreif) [inline Changelog.md:11]:
```suggestion
      When used to modify the `actor` keyword in an actor or actor class definition, the keyword declares that the default stability of a
```

File: Changelog.md [hunk 11]
```diff
@@ -1,5 +1,40 @@
...
+        }
+
+      }
+      ```
+
+      On upgrade, the transient variable `invocations` will be reset to `0` and `value`, now implicitly `stable`, will retain its current value.
+
+      Legacy actors and classes declared without the `persistent` keyword have the same semantics as before.
+
+
```
Comment (ggreif) [inline Changelog.md:37]:
```suggestion

```

**Review by ggreif — APPROVED** [review]
Two minor things, otherwise LGTM!

**Review by ggreif — APPROVED** [review]

**Comment by ggreif** [comment]
@Mergifyio update

**Pull request merged**